### ISSUE TITLE
OpenNI grabber blocks for 1 second on startup

### ISSUE THREAD

**Issue by nh2** [PointCloudLibrary/pcl#682]
See my comment at https://github.com/PointCloudLibrary/pcl/commit/0599508c#commitcomment-6362676

**Comment by kunaltyagi** [comment]
No PR yet...

[Issue is still present](https://github.com/PointCloudLibrary/pcl/blob/master/io/src/openni_grabber.cpp#L254)

**Comment by SergioRAgostinho** [comment]
Why would we want to read a frame if it's corrupted? Would it not be better to force pulling it from the sensor, discard it and only then implement the continuous grabbing loop?

**Comment by kunaltyagi** [comment]
> force pulling it from the sensor, discard it and only then implement the continuous grabbing loop

Yeah, that's what I meant by "read", except I don't know if it's even possible to "force pull" a frame. Eg: the length might be messed up.
